File: RPGhtml/Script/Var.js
var canvas;
var context;
var canvasRext;
var gameInstance;

var scene = 'Title';
var state = '';

var select = {
    roguelikeLevel : -1,
    roguelikeCharacter : -1,
    numOfCard : 0,
};

var pool = {
    card : {
        'Fire' : [],
        'Water' : [],
        'Nature' : [],
        'Earth' : [],
        'Light' : [],
        'Dark' : [],
        'Normal' : [],
    },

    equip : {
        'Fire' : [],
        'Water' : [],
        'Nature' : [],
        'Earth' : [],
        'Light' : [],
        'Dark' : [],
        'Normal' : [],
    },

    skill : {
        'Fire' : [],
        'Water' : [],
        'Nature' : [],
        'Earth' : [],
        'Light' : [],
        'Dark' : [],
        'Normal' : [],
    },
};

var field = {
    place : '',
    floor : [],
    wall : [],
    thing : [],
    portal : [],

    camera : [0, 0],
    playerPosition : [0, 0]
};

var player = {
    level : 0,
    exp : 0,
    expMax : 0,
    life : 0,

    card : [],
    equip : [],
    item : [],
    deck : [],
    techTree : [],
};

var varRoguelike = {
    floor : 0,
    map : [[0, 1, 1, 1, 1],
           [1, 1, 1, 1, 1],
           [1, 1, 1, 1, 1],
           [1, 1, 1, 1, 1],
           [1, 1, 1, 1, 5]],
};

var playerRoguelike = {
    playerID : 0,
    playerType : '',
    level : 0,
    exp : 0,
    expMax : 0,
    life : 0,
    skill : 0,

    deck : [],
    equip : [],
    item : [],
    techTree : [],
};

File: RPGpycli/Script/var.py
window = None
scene = 'title'
state = ''

class Cursor():
    class Title():
        menu = 0

    class Save_Select():
        save = 0

    class Roguelike_Menu():
        level = 0
        character = 0

    class Roguelike_Start():
        select = 0

class Save():
    data = {
        0 : '',
        1 : '',
        2 : '',
    }

class Select():
    class Roguelike_Menu():
        level = 0
        character = 0

class Pool():
    card = { 
        'all' : [],
        'fire' : [],
        'water' : [],
        'nature' : [],
        'earth' : [],
        'light' : [],
        'normal' : []
    }

    item = {
        'common' : [],
    }

class Field():
    wall = []
    thing = []
    portal = []

class Player():
    level = 0
    exp = 0
    exp_max = 0
    life = 20

    card = []
    equip = []
    item = []

class Player_Rogue():
    level = 0
    exp = 0
    exp_max = 0
    life = 20

    deck = []
    equip = []
    item = []
